### ISSUE TITLE
middy-jsonapi treats BadRequest errors from Middy validation as unhandled errors

### ISSUE THREAD

**Issue by kpeters-cbsi** [willfarrell/middy-jsonapi#6]
I have a Typescript Lambda set up like so:

```
import { APIGatewayProxyHandler } from 'aws-lambda'
import middy from '@middy/core'
import cors from '@middy/http-cors'
import httpJsonBodyParser from '@middy/http-json-body-parser'
import jsonapi from 'middy-jsonapi'
import validator from '@middy/validator'
import httpContentNegotiation from '@middy/http-content-negotiation'

const main: APIGatewayProxyHandler = async (event) => { 
  ...
}

export const handler = middy(main)
  .use(httpJsonBodyParser())
  .use(cors())
  .use(jsonapi({ response }))
  .use(
    httpContentNegotiation({
      availableLanguages: ['en-US'],
      availableMediaTypes: ['application/vnd.api+json'],
    })
  )
  .use(validator({ inputSchema }))
```

But when I send an invalid payload to it, I get a 500 back instead of a 400. I've traced the error to [line 72 of index.js](https://github.com/willfarrell/middy-jsonapi/blob/9e607efd81968552c63cb03d568e3a6e66b730b4/index.js#L72). The error class is `BadRequestError`, but `handler.error instanceof createError.BadRequest` evaluates to false. Not sure where to take it from there, otherwise I'd whip up a PR.

**Comment by willfarrell** [comment]
I'm open to reviewing a PR

**Comment by kpeters-cbsi** [comment]
You would say that. 🤣 Let me see what I can do.

One thing I notice off the bat is that it doesn't look like `validator` is using `http-errors` ([link](https://github.com/middyjs/middy/blob/96b2ce63059b43bd27040d4a8ba45352c24f67c2/packages/validator/index.js#L42)):

```
      const { createError } = require('@middy/util')
      ...
      const error = createError(400, 'Event object failed validation')
```

`createError` is defined in [@middy/util](https://github.com/middyjs/middy/blob/96b2ce63059b43bd27040d4a8ba45352c24f67c2/packages/util/index.js#L192)

So you might, for example, be able to check for `error.status && error.status === 400`

**Comment by willfarrell** [comment]
The createError function within `@middy/util` is a stripped down version of `http-errors` to keep the middleware as small as possible. It's the same object, just without all the developer helper features.

**Comment by kpeters-cbsi** [comment]
Right, but it's not the same object. That's why the check is failing.

**Comment by willfarrell** [comment]
If that's the case would you like to submit a PR to @middy/util with some additional tests? I can PR review other there :).

**Comment by kpeters-cbsi** [comment]
No, I think @middy/util is fine. If they want to use a stripped down version of `http-errors` that's their business. I think the change needs to happen in this codebase.
